aggregator/src/aggregator.rs

```diff
@@ -3325,6 +3355,38 @@ impl VdafOps {
     }
 }
 
+fn write_task_aggregation_counter<C: Clock>(
+    datastore: Arc<Datastore<C>>,
+    shard_count: u64,
+    task_id: TaskId,
+    counters: TaskAggregationCounter,
+) {
+    // We write task aggregation counters back in a separate tokio task & datastore transaction,
+    // so that any slowness induced by writing the counters (e.g. due to transaction retry) does
+    // not slow the main processing. The lack of transactionality between writing the updated
+    // aggregation job & updating the counters means that process failure may cause us to leave
+    // some counter updates unaccounted for, but that is an acceptable tradeoff.
+    let ord = thread_rng().gen_range(0..shard_count);
+    tokio::task::spawn(async move {
```

Comment:
I wonder if we should `spawn()` in the report writer too https://github.com/divviup/janus/blob/main/aggregator/src/aggregator/report_writer.rs#L201. (not in this PR, ofc)

Reply:
I agree that `spawn` isn't warranted for the report-counters case, for the reason that David points out. (We may well want to swap the ordering, since one of the hoped-for purposes of writing the counters separately from the reports is that the counters' write latency won't be reflected onto the report upload.)

If anything, I think the task aggregation counters might eventually want a solution more like report writer's, e.g. a background task that receives updates, then batches & writes them as a unit. I'm mildly concerned that if the DB becomes unhealthy, this could lead to a bunch of unmanaged tasks hanging out & trying to write counters -- the background task would ensure that there is only one counter-writing task, I suppose.